### PR TITLE
Fix platform from shutting down immediately on develop

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description

This PR fixes the issue where the carma-platform shuts down or vehicle not moving in cdasim. 
- carma-platform shuts down due to spat processing at the moment. There are more PRs coming to correct the spat processing, but this lets it not shut down
- intersection_transit_maneuvering was not correctly generating trajectory despite being able to call the inalencruising_plugin. 
- Fixed unit test for intersection_transit_maneuvering packages as well.
- This is because the shared_ptr was not passed by reference, so when it goes out of scope, the shared_ptr was being deleted
- Vehicle was not moving cdasim because the localization stack was printing on top of the CARLA's pose printing with a different timestamp which was freezing or jumping the car ahead in position.
## Related GitHub Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/2177
https://github.com/usdot-fhwa-stol/carma-platform/issues/2168

## Related Jira Key
[CDAR-391](https://usdot-carma.atlassian.net/browse/CDAR-391)

## Motivation and Context
updating the cdasim environment to develop

## How Has This Been Tested?
tested in cdasim voices2 computer

## Types of changes

- [X] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

[CDAR-391]: https://usdot-carma.atlassian.net/browse/CDAR-391?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: intersection_transit_maneuvering/include/intersection_transit_maneuvering/call_interface.hpp [hunk 1]
```diff
@@ -23,7 +23,7 @@
 class CallInterface
 {
     public:
-        virtual void call(carma_planning_msgs::srv::PlanTrajectory::Request::SharedPtr req, carma_planning_msgs::srv::PlanTrajectory::Response::SharedPtr resp) = 0;
+        virtual void call(carma_planning_msgs::srv::PlanTrajectory::Request::SharedPtr req, carma_planning_msgs::srv::PlanTrajectory::Response::SharedPtr& resp) = 0;
```
Comment (saina-ramyar) [inline intersection_transit_maneuvering/include/intersection_transit_maneuvering/call_interface.hpp:26]: probably not as important, but shouldt the req be also passed by reference?

File: intersection_transit_maneuvering/src/intersection_transit_maneuvering.cpp [hunk 2]
```diff
@@ -197,17 +197,19 @@ return CallbackReturn::SUCCESS;
             new_req->maneuver_plan.maneuvers.push_back(i);
         }
 
-            new_req->header = req->header;
-            new_req->vehicle_state = req->vehicle_state;
-            new_req->initial_trajectory_plan = req->initial_trajectory_plan;
-            object_->call(new_req,resp);
```
Comment (saina-ramyar) [inline intersection_transit_maneuvering/src/intersection_transit_maneuvering.cpp:203]: I dont see the "resp" definition being removed after you have renamed it?!

File: intersection_transit_maneuvering/include/intersection_transit_maneuvering/call_interface.hpp (hunk 1 above)
Comment (MishkaMN) [reply intersection_transit_maneuvering/include/intersection_transit_maneuvering/call_interface.hpp:26]: Req is not being changed, so we probably shouldn't pass them by reference

File: intersection_transit_maneuvering/src/intersection_transit_maneuvering.cpp (hunk 2 above)
Comment (MishkaMN) [reply intersection_transit_maneuvering/src/intersection_transit_maneuvering.cpp:203]: resp is the input to the callback, so the definition shouldn't be removed. 
the callback is still using the resp variable in line 212 though from ilc_resp to resp

**Review by saina-ramyar — APPROVED** [review]

**Pull request merged**